packages/organizations/origin-organization-irec-api/src/irec/irec.mock.service.ts

```diff
@@ -65,110 +127,67 @@ export class IrecMockService implements IIrecService {
-            countryCode: 'GB',
-            defaultAccount: 'someTradeAccount',
-            deviceType: 'TC110',
-            fuelType: 'ES200',
-            issuer: 'someIssuerCode',
-            latitude: '53.405088',
-            longitude: '-1.744222',
-            name: 'DeviceXYZ',
-            notes: 'Lorem ipsum dolor sit amet',
-            registrantOrganization: 'someRegistrantCode',
-            registrationDate: new Date('2001-09-20'),
-            status: DeviceState.Approved,
-            code: 'mockDeviceCode',
-            active: true
-        };
+        return this.devices.find((d) => d.code === code);
     }
 
-    async getDevices(user: UserIdentifier): Promise<IrecDevice[]> {
-        return [
-            {
-                address: '1 Wind Farm Avenue, London',
-                capacity: 500,
-                commissioningDate: new Date('2001-08-10'),
-                countryCode: 'GB',
-                defaultAccount: 'someTradeAccount',
-                deviceType: 'TC110',
-                fuelType: 'ES200',
-                issuer: 'someIssuerCode',
-                latitude: '53.405088',
-                longitude: '-1.744222',
-                name: 'DeviceXYZ',
-                notes: 'Lorem ipsum dolor sit amet',
-                registrantOrganization: 'someRegistrantCode',
-                registrationDate: new Date('2001-09-20'),
-                status: DeviceState.Approved,
-                code: 'mockDeviceCode',
-                active: true
-            }
-        ];
+    async getDevices(user: UserIdentifier): Promise<Device[]> {
+        return this.devices;
     }
 
     async getAccountInfo(user: UserIdentifier): Promise<Account[]> {
-        return [
-            {
-                code: 'TEST001',
-                details: {
-                    name: 'Some new revision',
-                    private: false,
-                    restricted: false,
-                    active: true,
-                    notes: 'Some test'
-                },
-                type: AccountType.Trade
-            },
-            {
-                code: 'TESTREDEMPTIONACCOUNT',
-                details: {
-                    name: 'Test Account Details 001',
-                    private: false,
-                    restricted: false,
-                    active: true,
-                    notes: 'Some test notes'
-                },
-                type: AccountType.Redemption
-            }
-        ];
+        return this.accountInfo;
     }
 
     async getTradeAccountCode(user: UserIdentifier): Promise<string> {
         const accounts = await this.getAccountInfo(user);
         return accounts.find((account: Account) => account.type === AccountType.Trade)?.code || '';
     }
 
+    async getIssueAccountCode(user: UserIdentifier): Promise<string> {
+        const accounts = await this.getAccountInfo(user);
+        return accounts.find((account: Account) => account.type === AccountType.Issue)?.code || '';
```

Comment:
So this can return empty string - is it properly handled later on or will randomly throw if empty string is returned?

Usually one may consider either throwing or returning some different type (for example null, if strict null checks are enabled) to ensure, that this "fallback value" is properly handled.

Reply:
It is handled later